lib/src/main/java/com/auth0/android/lock/views/SocialView.java

```diff
@@ -75,6 +76,15 @@ private void init(boolean smallButtons) {
         setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
     }
 
+    private List<AuthConfig> generateAuthConfigs(List<Strategy> socialStrategies) {
+        List<AuthConfig> configs = new ArrayList<>();
+        for (Strategy s : socialStrategies) {
+            int style = lockWidget.getConfiguration().authStyleForConnection(s.getName(), s.getDefaultConnectionName());
+            configs.add(new AuthConfig(s.getName(), s.getDefaultConnectionName(), style));
```

Comment:
`authStyleForConnection` uses AuthConfig for default values, and here we create again. Why don't just have the array of `AuthConfig` on `Configuration`? And the `AuthConfig` could have a constructor without custom style that would use the default (so the static method can be private)